max-rps/main.go

```diff
@@ -0,0 +1,231 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"math"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+	pb "github.com/buoyantio/strest-grpc/protos"
+
+	"golang.org/x/net/context"
+	"gonum.org/v1/gonum/matrix/mat64"
+	"gonum.org/v1/gonum/optimize"
+	"google.golang.org/grpc"
+)
+
+// `strest-max-rps` is designed to tell you the maximum rps that
+// either a strest-grpc server or an intermediary can provide. It does
+// this using the Universal Scalability Law.
+//
+// Thanks to @brendantracey for the go playground snippet least squared regression
+// code that I borrowed verbatim.
+func main() {
+	var (
+		address           = flag.String("address", "localhost:11111", "hostname:port of strest-grpc service or intermediary")
+		concurrencyLevels = flag.String("concurrencyLevels", "1,5,10,20,30", "levels of concurrency to test with")
+		timePerLevel      = flag.Duration("timePerLevel", 1 * time.Second, "how much time to spend testing each concurrency level")
+		debug             = flag.Bool("debug", false, "print out some extra information for debugging")
+	)
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "Usage: %s [flags]\n", path.Base(os.Args[0]))
+		flag.PrintDefaults()
+	}
+
+	flag.Parse()
+
+	if *timePerLevel < time.Second {
+		log.Fatalf("timePerLevel cannot be less than 1 second.")
+	}
+
+	levels := strings.Split(*concurrencyLevels, ",")
+	var denseLatency [](float64)
+
+	for _, l := range levels {
+		level, err := strconv.Atoi(l)
+		if err != nil {
+			log.Fatalf("unknown concurrency level: %s", l, err)
```

Comment:
Looks like you're missing a second placeholder in your string here. I think it should be:

```
log.Fatalf("unknown concurrency level: %s, %s", l, err)
```

Or similar.